django_admin_listfilter_dropdown/templates/admin/change_list.html

```diff
@@ -0,0 +1,52 @@
+{% extends "admin/change_list.html" %}
+{% load staticfiles %}
+
+{% block extrahead %}
+    <script type="text/javascript" src="{% static 'admin/js/vendor/jquery/jquery.js' %}"></script>
+    <script type="text/javascript" src="{% static 'admin/js/vendor/select2/select2.full.js' %}"></script>
+
+    <script>
+        (function () {
+            var init = function ($element, options) {
+                var settings = $.extend({
+                    ajax: {
+                        data: function (params) {
+                            return {
+                                term: params.term,
+                                page: params.page
+                            };
+                        }
+                    }
+                }, options);
+                $element.select2(settings).on('change', function (ev) {
```

Comment:
currently when the remove icon is clicked, the select2 dropdown opens and is visible for a short time before the page reloads. This is not beautiful. Using select2's `select2:unselect` doesnt help. Select2 allows to deactivate the remove icon and such functionality could be implemented separately on the right of the select2 box.